### ISSUE TITLE
compare.LessThan not work rightly, when compare two very big int/int64 number

### ISSUE THREAD

**Issue by duanlianjie** [duke-git/lancet#274]
Hi, here is a bug.

when I use compare.LessThan func to compare two very big int number, for example 645680099112988673 and 645680099112988675, in this func internal, it will transfer both number to float64 type, all is 645680099112988700, see picture below, finaly it return false, it not correct.

```golang
func TestCompareLessThan(t *testing.T) {
	tests := []struct {
		name string
		args [2]any
		want bool
	}{
		{
			name: "test1",
			args: [2]any{645680099112988673, 645680099112988675},
			want: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			t.Log(tt.args[0].(int) < tt.args[1].(int))
			if got := compare.LessThan(tt.args[0], tt.args[1]); got != tt.want {
				t.Errorf("LessThan() = %v, want %v", got, tt.want)
			}
		})
	}
}

```

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/edf22976-9e77-433f-b7ed-7cf369a6abd0">

**Comment by duanlianjie** [comment]
my env
GOOS='darwin'
GOARCH='arm64'
GOVERSION='go1.21.1'

lancet version
v2.2.8

**Comment by duke-git** [comment]
@duanlianjie, this issue was fixed in rc branch. will release in v2.3.4.

**Comment by duke-git** [comment]
fixed in v2.3.4
